src/settings/SettingsStore.js

```diff
@@ -0,0 +1,462 @@
+/*
+Copyright 2017 Travis Ralston
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import Promise from 'bluebird';
+import DeviceSettingsHandler from "./DeviceSettingsHandler";
+import RoomDeviceSettingsHandler from "./RoomDeviceSettingsHandler";
+import DefaultSettingsHandler from "./DefaultSettingsHandler";
+import RoomAccountSettingsHandler from "./RoomAccountSettingsHandler";
+import AccountSettingsHandler from "./AccountSettingsHandler";
+import RoomSettingsHandler from "./RoomSettingsHandler";
+import ConfigSettingsHandler from "./ConfigSettingsHandler";
+import {_t, _td} from '../languageHandler';
+import SdkConfig from "../SdkConfig";
+
+// Preset levels for room-based settings (eg: URL previews).
+// Doesn't include 'room' because most settings don't need it. Use .concat('room') to add.
+const LEVELS_PRESET_ROOM = ['device', 'room-device', 'room-account', 'account', 'config'];
```

Comment:
As per #riot-dev, the naming feels wrong here - especially that LEVELS_PRESET_ROOM doesn't include the 'room' granularity(!)

How about having:
 * LEVELS_ROOM_SETTINGS <-- to match the old RoomSettings stuff
 * LEVELS_ROOM_SETTINGS_WITH_ROOM <-- to match this + 'room'
 * LEVELS_ACCOUNT_SETTINGS
 * LEVELS_FEATURES

I think the 'presets' are redundant as long as the consts are labelled correctly?

(N.B. i don't think this is perfect, but anything to kill the confusing `.concat('room')` stuff...)